application/src/main/java/org/togetherjava/tjbot/features/help/HelpSystemHelper.java

```diff
@@ -131,6 +143,90 @@ private RestAction<Message> sendExplanationMessage(GuildMessageChannel threadCha
+     * @param threadChannel - The thread in which the question was asked.
+     * @return An answer for the user from the AI service or a message indicating either an error or
+     *         why the message wasn't used.
+     */
+    RestAction<Message> constructChatGptAttempt(ThreadChannel threadChannel,
+            String questionFirstMessage) {
+
+        Optional<String> questionOptional =
+                prepareChatGptQuestion(threadChannel, questionFirstMessage);
+        Optional<String> chatGPTAnswer;
+
+        try {
+            String question = questionOptional.orElseThrow();
+            logger.debug("The final question sent to chatGPT: {}", question);
+            chatGPTAnswer = chatGptService.ask(question);
+
+            String aiResponse = chatGPTAnswer.orElseThrow();
+            String lowercaseAiResponse = aiResponse.toLowerCase(Locale.US);
+            if (lowercaseAiResponse.contains("as an ai language model")) {
+                logger.debug("Response from ChatGPT, lacks context or unable to respond?: {}",
+                        aiResponse);
+                return useChatGptFallbackMessage(threadChannel);
+            }
+
+            UnaryOperator<String> response =
+                    """
+                            Here is an AI assisted attempt to answer your question 🤖. Maybe it helps! \
+                            In any case, a human is on the way 👍. To continue talking to the AI, you can use \
+                            %s.
+                            """::formatted;
+
+            return mentionGuildSlashCommand(threadChannel.getGuild(), ChatGptCommand.COMMAND_NAME)
+                .map(response)
+                .flatMap(threadChannel::sendMessage)
+                .flatMap(embed -> threadChannel
+                    .sendMessageEmbeds(HelpSystemHelper.embedWith(aiResponse)));
+        } catch (NoSuchElementException e) {
+            logger.warn("Something went wrong while trying to communicate with the ChatGpt API: {}",
+                    e.getMessage());
+            return useChatGptFallbackMessage(threadChannel);
+        }
+    }
+
+    private Optional<String> prepareChatGptQuestion(ThreadChannel threadChannel,
+            String questionFirstMessage) {
+        String questionTitle = threadChannel.getName();
+        StringBuilder questionBuilder = new StringBuilder(MAX_QUESTION_LENGTH);
+
+        if (questionFirstMessage.length() < MIN_QUESTION_LENGTH
+                && questionTitle.length() < MIN_QUESTION_LENGTH) {
+            return Optional.empty();
+        }
+
+        if (questionTitle.length() + questionFirstMessage.length() > MAX_QUESTION_LENGTH) {
+            // - 1 for the space that will be added afterward.
+            questionFirstMessage = questionFirstMessage.substring(0,
+                    (MAX_QUESTION_LENGTH - 1) - questionTitle.length());
+        }
+
+        questionBuilder.append(questionTitle).append(" ").append(questionFirstMessage);
```

Comment:
i think it would be easier to read if the order would be changed here. like:
```java
questionBuilder.append(title).append(" ");
String shortQuestion = question.substring(0, questionBuilder.length());
questionBuilder.append(shortQuestion);
```
Then the comment is also not needed anymore, since its understandable.

U have to see how that `if` fits into the mix though.

Reply:
```
questionBuilder.append(questionTitle).append(" ");
        if (originalQuestion.length() > MAX_QUESTION_LENGTH - questionBuilder.length()) {
            originalQuestion = originalQuestion.substring(0, MAX_QUESTION_LENGTH - questionBuilder.length());
        }

        questionBuilder.append(originalQuestion);
```

This is what I got after some refactoring. More or less readable?